pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -107,12 +108,33 @@ func add(mgr manager.Manager, r *DirectReconciler, reconcilePredicate predicate.
 		predicateList = append(predicateList, reconcilePredicate)
 	}
 
+	secretObj := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "Secret",
+			"apiVersion": "v1",
+		},
+	}
+
 	_, err := builder.
 		ControllerManagedBy(mgr).
 		Named(r.controllerName).
 		WithOptions(crcontroller.Options{MaxConcurrentReconciles: k8s.ControllerMaxConcurrentReconciles, RateLimiter: ratelimiter.NewRateLimiter()}).
 		WatchesRawSource(&source.Channel{Source: r.immediateReconcileRequests}, &handler.EnqueueRequestForObject{}).
 		For(obj, builder.OnlyMetadata, builder.WithPredicates(predicateList...)).
+		Watches(
```

Comment:
Does this mean that all direct controllers will watch the Secret? Is it possible to check the input `DirectReconciler` first before adding the new watch?

Reply:
Good point! Done.